### PR TITLE
Update Microsoft.Extensions packages 

### PR DESCRIPTION
**Pull request by ryuyu**
Update references to Microsoft.Extensions packages from 1.x -> 2.x
Also retargets net472

References to packages will be updated with official builds.
Depends on https://github.com/NuGet/ServerCommon/pull/318

### CODE REVIEW

File: src/AccountDeleter/AccountDeleter.csproj [hunk 1]
```diff
@@ -106,6 +106,23 @@
...
+      <Version>2.58.0</Version>
+    </PackageReference>
+    <PackageReference Include="NuGet.Services.Messaging.Email">
+      <Version>2.58.0</Version>
+    </PackageReference>
+    <PackageReference Include="NuGet.Services.ServiceBus">
+      <Version>2.58.0</Version>
+    </PackageReference>
+    <PackageReference Include="NuGet.Services.Validation.Common.Job">
+      <Version>4.3.0-ryuyu-update-extensions-3142787</Version>
```
Comment (joelverhagen) [inline src/AccountDeleter/AccountDeleter.csproj]: missing a closing `</PackageReference>`

File: src/NuGetGallery/Web.config [hunk 2]
```diff
@@ -501,297 +505,81 @@
...
-        <assemblyIdentity name="Autofac.Integration.Owin" publicKeyToken="17863AF14B0044DA" culture="neutral"/>
-        <bindingRedirect oldVersion="0.0.0.0-4.2.0.0" newVersion="4.2.0.0"/>
-      </dependentAssembly>
-      <dependentAssembly>
-        <assemblyIdentity name="Autofac" publicKeyToken="17863AF14B0044DA" culture="neutral"/>
-        <bindingRedirect oldVersion="0.0.0.0-4.6.2.0" newVersion="4.6.2.0"/>
-      </dependentAssembly>
-    </assemblyBinding>
-  </runtime>
+  	<assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
```
Comment (joelverhagen) [inline src/NuGetGallery/Web.config]: Did we regenerate this whole thing?

nit: fix indentation

File: src/AccountDeleter/AccountDeleter.csproj [hunk 3]
```diff
@@ -106,6 +106,23 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
+    <PackageReference Include="NuGet.Jobs.Common">
```
Comment (skofman1) [inline src/AccountDeleter/AccountDeleter.csproj]: why do we need to add those dependencies directly? Don't we get the transitively from NuGet.Services.Validation.Common.Job?

File: src/DatabaseMigrationTools/DatabaseMigrationTools.csproj [hunk 4]
```diff
@@ -66,13 +63,14 @@
...
+    <PackageReference Include="NuGet.Jobs.Common">
+      <Version>4.3.0-ryuyu-update-extensions-3142787</Version>
     </PackageReference>
-    <PackageReference Include="Microsoft.Extensions.Options">
-      <Version>1.1.2</Version>
+    <PackageReference Include="NuGet.Services.Configuration">
+      <Version>2.58.0</Version>
     </PackageReference>
-    <PackageReference Include="NuGet.Jobs.Common">
+    <PackageReference Include="NuGet.Services.Validation">
```
Comment (skofman1) [inline src/DatabaseMigrationTools/DatabaseMigrationTools.csproj]: NuGet.Services.Validation dependency appears twice..

File: src/DatabaseMigrationTools/DatabaseMigrationTools.csproj [hunk 5]
```diff
@@ -66,13 +63,14 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
-    <PackageReference Include="Microsoft.Extensions.Logging">
-      <Version>1.1.2</Version>
+    <PackageReference Include="NuGet.Jobs.Common">
```
Comment (skofman1) [inline src/DatabaseMigrationTools/DatabaseMigrationTools.csproj]: why are new dependencies needed?

File: src/NuGet.Services.DatabaseMigration/NuGet.Services.DatabaseMigration.csproj [hunk 6]
```diff
@@ -76,19 +73,14 @@
...
-      <Version>1.1.2</Version>
-    </PackageReference>
-    <PackageReference Include="Microsoft.Extensions.Options">
-      <Version>1.1.2</Version>
-    </PackageReference>
     <PackageReference Include="NuGet.Build.Tasks.Pack">
       <Version>4.8.0</Version>
     </PackageReference>
     <PackageReference Include="NuGet.Jobs.Common">
+      <Version>4.3.0-ryuyu-update-extensions-3142787</Version>
```
Comment (skofman1) [inline src/NuGet.Services.DatabaseMigration/NuGet.Services.DatabaseMigration.csproj]: what's the plan to get a stable version here?

File: src/NuGetGallery/NuGetGallery.csproj [hunk 7]
```diff
@@ -2076,6 +2076,7 @@
       <Version>2.2.0</Version>
     </PackageReference>
     <PackageReference Include="NuGet.Services.Licenses">
+      <Version>2.58.0</Version>
       <Version>2.57.0</Version>
```
Comment (skofman1) [inline src/NuGetGallery/NuGetGallery.csproj]: the version tag appears twice..

File: src/NuGetGallery.Core/NuGetGallery.Core.csproj [hunk 8]
```diff
@@ -239,6 +239,16 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="NuGet.Services.FeatureFlags">
+      <Version>2.58.0</Version>
+    </PackageReference>
+    <PackageReference Include="NuGet.Services.Messaging.Email">
+      <Version>2.58.0</Version>
+    </PackageReference>
```
Comment (skofman1) [inline src/NuGetGallery.Core/NuGetGallery.Core.csproj]: why do we need those new dependencies?

File: src/NuGetGallery.Services/NuGetGallery.Services.csproj [hunk 9]
```diff
@@ -292,6 +292,10 @@
       <Version>5.0.0-preview1.5665</Version>
     </PackageReference>
     <PackageReference Include="NuGet.Services.KeyVault">
+      <Version>2.58.0</Version>
+    </PackageReference>
+    <PackageReference Include="NuGet.Services.Logging">
```
Comment (skofman1) [inline src/NuGetGallery.Services/NuGetGallery.Services.csproj]: the dependency is duplicated..

File: src/AccountDeleter/AccountDeleter.csproj (hunk 3 above)
Comment (ryuyu) [reply src/AccountDeleter/AccountDeleter.csproj]: Yes we do. Excellent point. removed.

File: src/NuGet.Services.DatabaseMigration/NuGet.Services.DatabaseMigration.csproj (hunk 6 above)
Comment (ryuyu) [reply src/NuGet.Services.DatabaseMigration/NuGet.Services.DatabaseMigration.csproj]: Plan was to check in Gallery -> get Gallery.Core build -> consume this in Jobs -> get Jobs.Common build -> update Gallery with new Jobs.Common.

File: src/NuGetGallery.Core/NuGetGallery.Core.csproj (hunk 8 above)
Comment (ryuyu) [reply src/NuGetGallery.Core/NuGetGallery.Core.csproj]: Turns out these are not new, they were supposed to be updates that got clobbered by the dev merge. I'm cleaning this up now.

File: src/AccountDeleter/AccountDeleter.csproj [hunk 10]
```diff
@@ -106,8 +106,20 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
+    <PackageReference Include="NuGet.Services.Configuration">
+      <Version>2.58.0</Version>
```
Comment (skofman1) [inline src/AccountDeleter/AccountDeleter.csproj]: remove?

Comment (ryuyu) [reply src/AccountDeleter/AccountDeleter.csproj]: Yes you are totally right. Removed.

**Pull request merged**